modules/siddhi-extensions/math/src/main/java/org/wso2/siddhi/extension/math/PercentileFunctionExtension.java

```diff
@@ -208,59 +277,56 @@ public void restoreState(Object[] state) {
 
         private final Attribute.Type type = Attribute.Type.DOUBLE;
         private long count = 0;
-        private List<Float> values = new ArrayList<Float>();
-        private DescriptiveStatistics stat = new DescriptiveStatistics();
+        private List<Double> valuesList = new ArrayList<Double>();
 
+        @Override
         public Attribute.Type getReturnType() {
             return type;
         }
 
         @Override
         public Object processAdd(Object[] data) {
             count++;
-            values.add((Float) data[0]);
             if (count == 0) {
                 return 0.0;
             }
 
-            stat.clear();
-            for (float value : values) {
-                stat.addValue(value);
-            }
-            return stat.getPercentile(percentileValue);
+            double value = (Float) data[0];
+
```

Comment:
Thanks for pointing it out. Those are already removed in another PR - https://github.com/wso2/siddhi/pull/189/files#diff-d8117cc8e8809a20ba580988248755e6L295